### ISSUE TITLE
Unable to specify ImagePullSecrets for private registry in ClusterServiceVersion

### ISSUE THREAD

**Issue by secat** [operator-framework/operator-lifecycle-manager#1294]
I have generated a CSV for my custom operator using the `operator-sdk generate csv` command. After inspecting the CRD definition for CSV (ClusterServiceVersion) custom resource, I didn't find a way to pass a configuration for `ImagePullSecrets` for the `clusterserviceversion.spec.install.spec.permissions.serviceAccountName`.

The image for my custom operator is in a private registry and I need to have an automated way to pass my private registry credentials.

Without proper `ImagePullSecrets` the pod created for my operator deployment by the OLM gets stuck with `ImagePullBackOff`.

Is there something that I can configure for that? Is it something that is currently missing?

I am currently using:
- operator-sdk: v0.15.0
- operator lifecycle manager: quay.io/operator-framework/olm@sha256:0d15ffb5d10a176ef6e831d7865f98d51255ea5b0d16403618c94a004d049373
- kubernetes: v1.15.5 (docker desktop)

**Comment by ecordell** [comment]
Secrets can be specified on the `CatalogSource` that packages up your operator for installation in the cluster.

```yaml
apiVersion: operators.coreos.com/v1alpha1
kind: CatalogSource
metadata:
  name: my-catalog
  namespace: default
spec:
  name: my-catalog
  sourceType: grpc
  image: <your catalog image>
   secrets:
  - my-pull-secret
```

Be aware that if you do this, the secret will be copied into whichever target namespace you install the operator.

Here's a link to an example (the commit removes, rather than adds, pull secrets: https://github.com/operator-framework/operator-lifecycle-manager/commit/0dc8e60f298a468f306d11f13dbd38489e8c573b)

**Comment by secat** [comment]
@ecordell thank you! 

It fixes the problem I was encountering.

**Comment by fantapsody** [comment]
Hello @ecordell and @secat , I just met the similar problem when I was trying to deploy my custom operator from GCR to minikube. However, the operator pod still got ErrImagePull after I specified the  name of the secret in the catalogsource manifests, while the same secret(created from a GCP service account) has been successfully used to pull the index image as the catalog source deployed in the same minikube cluster.

I also tried to search through the source code and try to find what happened under the hood, using `Secrets` or `ImagePullSecrets` as keywords. I wonder how it works after the `secrets` has been set, or could you please provide me some documents or source code or pull requests that I can dig into? Thanks! @ecordell

**Comment by jeyaramashok** [comment]
where should we create this image pull secret ? should this be created in the same namespace as the catalog source ?

**Comment by hassenius** [comment]
In my I'm not even seeing the secret being created in the target namespace. 
Some further instructions would be great

**Comment by sethpeterson** [comment]
I'm also not seeing my secret in the target namespace
@ecordell do you have any pointers on debugging this issue or any documentation on implementation?

**Comment by madorn** [comment]
experiencing the same issue here.

**Comment by geoberle** [comment]
I can see the same in Openshif 4.4.6

**Comment by prafull11** [comment]
I am not able to pull images from my private repo even when I provide secrets to catalogSource with openshift 4.5

**Comment by dasabhisek** [comment]
I faced the same issue. I set the image pull secrets under .spec.secrets of CatalogSource. But I am facing Error: ErrImagePull

Also this secrets are not copied into target namespace where Operator is getting installed.

**Comment by dol** [comment]
Inspired by https://github.com/operator-framework/operator-lifecycle-manager/pull/1878/files#diff-11246486fcec64bdb324ad92c24d1550098420a4c4add35a50f23706afa93ce4R113 the linking of the `default` service account with  `imagePullSecrets` the OLM namespace solved the issue.

`kubectl patch -n olm serviceaccount default -p '{"imagePullSecrets": [{"name": "my-pull-secret"}]}'`

**Comment by estroz** [comment]
Seems to be fixed by #1941.

@dol has suggested the correct workaround. In general, the service account that needs patching should be the one in the CatalogSource's namespace.
